dart/dynamics/FreeJoint.cpp

```diff
@@ -575,6 +556,16 @@ const std::string& FreeJoint::getStaticType()
 }
 
 //==============================================================================
+void FreeJoint::setTransformFromChildBodyNode(const Eigen::Isometry3d& T)
+{
+  Joint::setTransformFromChildBodyNode(T);
+
+  // We update Jacobian here since the Jacobian of FreeJoint is a function of
+  // "T" but not of joint positions.
+  mJacobian = math::getAdTMatrix(T);
```

Comment:
You are correct. I scanned all the `Joint`s for double check. I don't think we need to introduce the flag since it seems the consistency is not that important here.